components/script/dom/doge.rs

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+ use dom::bindings::cell::DOMRefCell;
+ use dom::bindings::codegen::Bindings::DogeBinding::{DogeMethods, DogeInit, Wrap as DogeWrap};
+ use dom::bindings::error::{Error, Fallible};
+ use dom::bindings::global::GlobalRef;
+ use dom::bindings::js::Root;
+ use dom::bindings::reflector::{Reflector, reflect_dom_object};
+ use dom::bindings::str::ByteString;
+ use rand;
+ use rand::Rng;
+ 
+ #[dom_struct]
+ pub struct Doge {
+     reflector_: Reflector,
+     such_list: DOMRefCell<Vec<ByteString>>,
+ }
+ 
+ impl Doge {
+     pub fn new_inherited() -> Doge {
+         Doge {
+             reflector_: Reflector::new(),
+             such_list: DOMRefCell::new(vec![]),
+         }
+     }
+ 
+     pub fn new(global: GlobalRef) -> Root<Doge> {
```

Comment:
A couple issues that I notice which we should update in our blog post:
* we now use `&GlobalScope` instead of `GlobalRef`
* we now use `DomRoot<Doge>` instead of `Root<Doge>`

Reply:
Actually, it looks like you're following an older revision of https://jeenalee.com/2016/10/03/implementing-doge-for-servo.html. What page were you using as a reference?

Reply:
Ok! Thanks @jdm. Now, we got it. We will resolve this and then we will write this in our blog. Just to keep a log of things that we are working on, we are writing our own team blog. :)